app/src/main/java/com/dmonzonis/nookpendium/MainActivity.kt

```diff
@@ -0,0 +1,151 @@
+        setUpFilterManager()
+        setUpSortManager()
+
+        toolbar.inflateMenu(R.menu.toolbar_menu)
+        setSupportActionBar(toolbar)
+        supportActionBar!!.apply {
+            setDisplayShowTitleEnabled(false)
+            setDisplayHomeAsUpEnabled(true)
+            setHomeAsUpIndicator(R.drawable.ic_menu_white_24dp)
+        }
+    }
+
+    private fun navListener() : NavigationView.OnNavigationItemSelectedListener {
+        return NavigationView.OnNavigationItemSelectedListener {
+            var selectedFragment : Fragment? = null
+
+            when (it.itemId) {
+                R.id.miGameAcnh -> {
+                    selectedFragment = CritterFragment(R.string.game_acnh)
+                    sharedPrefs.edit().putInt("selectedGame", R.string.game_acnh).apply()
+                }
+                R.id.miGameAcnl -> {
+                    selectedFragment = CritterFragment(R.string.game_acnl)
+                    sharedPrefs.edit().putInt("selectedGame", R.string.game_acnl).apply()
+                }
+                R.id.miAbout -> selectedFragment = AboutFragment()
+            }
+            if (selectedFragment != null) {
+                supportFragmentManager.beginTransaction()
+                    .replace(R.id.fragment_container, selectedFragment).commit()
+                if (selectedFragment is CritterFragment) {
+                    filterManager.fragment = selectedFragment
+                    sortManager.fragment = selectedFragment
+                }
+            }
+            drawerLayout.closeDrawer(navGamesView)
+            true
+        }
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
+        menuInflater.inflate(R.menu.toolbar_menu, menu)
+        // Set up search action
+        var searchMenuItem = menu?.findItem(R.id.miSearch)
+        val searchView = searchMenuItem?.actionView as SearchView
+        searchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
+            override fun onQueryTextSubmit(query: String?): Boolean {
+                return false  // Do nothing, search is done on text change
+            }
+
+            override fun onQueryTextChange(newText: String?): Boolean {
+                filterManager.searchQuery = newText ?: ""
+                filterManager.fragment.recomputeFilters()
+                return false
+            }
+
+        })
+        return super.onCreateOptionsMenu(menu)
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
+        closeAllDrawers()
+        when (item?.itemId) {
+            android.R.id.home -> drawerLayout.openDrawer(navGamesView)
+            R.id.miFilterButton -> drawerLayout.openDrawer(filterMenu)
+        }
+        return true
+    }
+
+    private fun closeAllDrawers() {
+        drawerLayout.closeDrawer(navGamesView)
+        drawerLayout.closeDrawer(filterMenu)
+    }
+
+    private fun addFilterField(id: Int, stringId: Int) {
```

Comment:
maybe this and all the methods below should be part of the critter fragment class, since they are only used there?